cmd/action/internal/update/update.go

```diff
@@ -0,0 +1,150 @@
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pelletier/go-toml/v2"
+	"golang.org/x/mod/semver"
+)
+
+type (
+	// Store is where the latest release data is stored on CLI host
+	Store struct {
+		Version   string    `toml:"version"`
+		URL       string    `toml:"url"`
+		CheckedAt time.Time `toml:"checkedat"`
+	}
+
+	// LatestRelease implements the required fields from https://api.github.com/repos/ariga/atlas/releases/latest
+	LatestRelease struct {
+		Version string `json:"tag_name"`
+		URL     string `json:"html_url"`
+	}
+)
+
+// CheckForUpdate implements a notification to the user when a later release is available
+// 1. Check release file ~/.local/atlas/release.toml for latest known release and poll time
+// 2. If last poll was more than 24h, poll GitHub public API https://docs.github.com/en/rest/reference/releases#get-the-latest-release
+// 3. Store the latest release metadata
+// 4. If current build Version, that is not development, is lower than the latest release, notify user
+func CheckForUpdate(version string, logF func(i ...interface{})) {
+	if !enabled(version) {
+		return
+	}
+	p, err := homedir.Expand("~/.config/atlas")
+	if err != nil {
+		return
+	}
+	ok, message, err := shouldUpdate(version, p, latestReleaseFromGithub)
+	if err != nil || !ok {
+		return
+	}
+	logF(message)
+}
+
+func enabled(version string) bool {
+	if _, ok := os.LookupEnv("ATLAS_NO_UPDATE_NOTIFIER"); ok {
+		return false
+	}
+	if _, ok := os.LookupEnv("GITHUB_ACTIONS"); ok {
+		return false
+	}
+	if ok := semver.IsValid(version); !ok {
+		return false
+	}
+	return true
+}
+
+func shouldUpdate(version string, path string, latestReleaseF func() (LatestRelease, error)) (bool, string, error) {
+	r, _ := localStore(path)
+	if shouldSkip(r) {
+		return false, "", nil
+	}
```

Comment:
Why are you ignoring the error here?

Reply:
if the file can't be read, we prefer to recreate it - we only "pay" with an extra polling to github api.